src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedPrimaryConstructor.cs

```diff
@@ -99,5 +118,55 @@ public IEnumerable<FieldSymbol> GetBackingFields()
             Interlocked.CompareExchange(ref _capturedParameters, Binder.CapturedParametersFinder.GetCapturedParameters(this), null);
             return _capturedParameters;
         }
+
+        internal override (CSharpAttributeData?, BoundAttribute?) EarlyDecodeWellKnownAttribute(ref EarlyDecodeWellKnownAttributeArguments<EarlyWellKnownAttributeBinder, NamedTypeSymbol, AttributeSyntax, AttributeLocation> arguments)
+        {
+            Debug.Assert(arguments.SymbolPart == AttributeLocation.Method);
+            arguments.SymbolPart = AttributeLocation.None;
+            var result = base.EarlyDecodeWellKnownAttribute(ref arguments);
+            arguments.SymbolPart = AttributeLocation.Method;
+            return result;
+        }
+
+        protected override void DecodeWellKnownAttributeImpl(ref DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments)
+        {
+            Debug.Assert(arguments.SymbolPart == AttributeLocation.Method);
+            arguments.SymbolPart = AttributeLocation.None;
+            base.DecodeWellKnownAttributeImpl(ref arguments);
+            arguments.SymbolPart = AttributeLocation.Method;
+        }
+
+        internal override void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttributeData> boundAttributes, ImmutableArray<AttributeSyntax> allAttributeSyntaxNodes, BindingDiagnosticBag diagnostics, AttributeLocation symbolPart, WellKnownAttributeData decodedData)
+        {
+            Debug.Assert(symbolPart is AttributeLocation.Method or AttributeLocation.Return);
+            base.PostDecodeWellKnownAttributes(boundAttributes, allAttributeSyntaxNodes, diagnostics, symbolPart is AttributeLocation.Method ? AttributeLocation.None : symbolPart, decodedData);
+        }
+
+        protected override bool ShouldBindAttributes(AttributeListSyntax attributeDeclarationSyntax, BindingDiagnosticBag diagnostics)
```

Comment:
nit: It is a bit odd to report diagnostics from a method called "ShouldXYZ" (wouldn't expect to have such side-effects).